packages/story-editor/src/components/library/panes/media/common/mediaGallery.js

```diff
@@ -44,39 +46,54 @@ const PHOTO_MARGIN = 4;
  * @return {*} The gallery element.
  */
 function MediaGallery({ resources, onInsert, providerType, canEditMedia }) {
-  const photos = resources.map((resource) => ({
-    id: resource.id,
+  const photos = resources.map((resource, index) => ({
+    index,
+    key: resource.id,
     src: resource.src,
     width: resource.width,
     height: resource.height,
   }));
 
+  const containerRenderer = ({ children }, ref) => {
+    return <GalleryContainer ref={ref}>{children}</GalleryContainer>;
+  };
+  const rowRenderer = ({ children }) => {
+    return children;
+  };
+
   const imageRenderer = useCallback(
-    ({ index, photo }) => (
-      <MediaElement
-        key={photo.id}
-        index={index}
-        margin={PHOTO_MARGIN + 'px'}
-        resource={resources[index]}
-        width={photo.width}
-        height={photo.height}
-        onInsert={onInsert}
-        providerType={providerType}
-        canEditMedia={canEditMedia}
-      />
-    ),
+    ({ photo, layout }) => {
+      return (
+        <MediaElement
+          key={photo.key}
+          index={photo.index}
+          margin={`0px 0px ${PHOTO_MARGIN}px 0px`}
+          resource={resources[photo.index]}
+          width={layout.width}
+          height={layout.height}
+          onInsert={onInsert}
+          providerType={providerType}
+          canEditMedia={canEditMedia}
+        />
+      );
+    },
     [providerType, onInsert, resources, canEditMedia]
   );
 
   return (
     <div>
       <Gallery
-        targetRowHeight={110}
-        direction={'row'}
-        // This should match the actual margin the element is styled with.
-        margin={PHOTO_MARGIN}
+        layout="rows"
         photos={photos}
-        renderImage={imageRenderer}
+        renderPhoto={imageRenderer}
+        renderRowContainer={rowRenderer}
+        renderContainer={forwardRef(containerRenderer)}
+        targetRowHeight={TARGET_ROW_HEIGHT}
+        rowConstraints={{
+          minPhotos: 2,
```

Comment:
With `minPhotos: 2` I still get 3 items in a row sometimes.

Ideally we only ever have 2. So `maxPhotos: 2`?